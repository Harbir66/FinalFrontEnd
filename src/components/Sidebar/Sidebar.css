.sidebar{
    width: 20vw;
    height: 100%;
    background: #272727;
    box-shadow: 0 2px 4px rgba(0,0,0,0.15);
    z-index: 100;
}
.sidebar-header{
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 1rem;
    height: 7vh;
    background: #5405c4;
    color: #fff;
    font-weight: 600;
    font-size: 3vh;
}

.sidebar-search{
    margin-top: 3vh;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 1rem;
    height: 7vh;
    color: #A1A1A1;
    font-weight: 600;
    font-size: 2vh;
}
.sidebar-search > img{
    height: 4vh;
}

.sidebar>nav{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    height: 40vh;
    color: #A1A1A1;
    font-weight: 600;
    font-size: 2vh;
}
.sidebar>nav>ul{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    padding: 0 1rem;
    height: 40vh;
    width: 80%;
    gap: 2rem;
    overflow-y: scroll;
    list-style-type: disc;
}
.sidebar>nav>ul>li{
    text-align: start;
    height: 3vh;
    width: 100%;
    cursor: pointer;
    color: #636363;
}

.sidebar>button{
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 4vh;
    background: inherit;
    color: #fff;
    font-weight: 600;
    font-size: 2vh;
    border: none;
    cursor: pointer;
}

.navlink{
    text-decoration: none;
    color: #fff;
    font-weight: 600;
    font-size: 2vh;
    width: 100%;
}