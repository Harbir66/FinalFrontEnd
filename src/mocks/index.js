export const mockCollection = [
  {
    id: 1,
    name: 'collection1',
    fields: 3,
    values: 3,
    allFields: ['field1', 'field2', 'field3'],
  },
  {
    id: 2,
    name: 'collection2',
    fields: 3,
    values: 3,
    allFields: ['field1', 'field2', 'field3'],
  },
];

export const mockHeading = 'Heading';

export const mockFields = ['field1', 'field2', 'field3'];
