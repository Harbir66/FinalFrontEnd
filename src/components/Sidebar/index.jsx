import React from 'react';
// import { useNavigate } from 'react-router-dom';
import './Sidebar.css';
import PropTypes from 'prop-types';
import searchIcon from '../../assets/icon-search-dark@3x.png';

function Sidebar({ collections, handleViewChange }) {
  const [selectedListItem, setSelectedListItem] = React.useState(null);
  const [builderPage, setBuilderPage] = React.useState(true);
  const handleListClick = (e) => {
    setSelectedListItem(e.target.innerText);
    setBuilderPage(false);
    handleViewChange('entries');
  };
  const handleBuildClick = () => {
    setBuilderPage(true);
    setSelectedListItem(null);
    handleViewChange('main');
  };
  return (
    <div className="sidebar">
      <div className="sidebar-header">CMS+</div>

      <div className="sidebar-search">
        <span>COLLECTION TYPES</span>
        <img src={searchIcon} alt="searchIcon" />
      </div>
      <nav>
        <ul>
          {collections &&
            collections.map((collection) => (
              <li
                style={{
                  background:
                    selectedListItem === collection.name ? 'black' : '',
                }}
                onClick={handleListClick}
                key={collection.id}
              >
                {collection.name}
              </li>
            ))}
        </ul>
      </nav>

      <button
        onClick={handleBuildClick}
        style={{ background: builderPage ? 'black' : '' }}
        type="button"
      >
        CONTENT TYPE BUILDER
      </button>
    </div>
  );
}

export default Sidebar;

Sidebar.propTypes = {
  collections: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  handleViewChange: PropTypes.func.isRequired,
};
