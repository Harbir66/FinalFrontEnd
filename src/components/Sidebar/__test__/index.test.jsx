import { render } from '@testing-library/react';
import React from 'react';
import { mockCollection } from '../../../mocks';
import Sidebar from '../index';

describe('Sidebar', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Sidebar
        collections={mockCollection}
        handleViewChange={jest.fn()}
        handleCollectionChange={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
