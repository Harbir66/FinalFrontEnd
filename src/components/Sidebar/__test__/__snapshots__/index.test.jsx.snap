// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <div
      class="sidebar-header"
    >
      CMS+
    </div>
    <div
      class="sidebar-search"
    >
      <span>
        COLLECTION TYPES
      </span>
      <img
        alt="searchIcon"
        src="icon-search-dark@3x.png"
      />
    </div>
    <nav>
      <ul>
        <li>
          collection1
        </li>
        <li>
          collection2
        </li>
      </ul>
    </nav>
    <button
      style="background: black;"
      type="button"
    >
      CONTENT TYPE BUILDER
    </button>
  </div>
</DocumentFragment>
`;
